src/main/java/com/minecolonies/util/StructureWrapper.java

```diff
@@ -275,141 +233,60 @@ private void placeBlock(IBlockState localState, @NotNull Block localBlock, @NotN
-                    }
-
-                    TileEntity tileEntity = world.getTileEntity(worldPos);
-                    saveTileEntity(schematic, tileEntity, minPos);
+                    return LanguageHandler.format("item.scepterSteel.scanFailure");
                 }
             }
+            catch(SecurityException e)
+            {
+                Log.getLogger().warn("Scan failed because of: ", e);
+                return LanguageHandler.format("item.scepterSteel.scanFailure");
+            }
         }
-        if (BlockPosUtil.isEqual(offset, 0, 0, 0))
-        {
-            offset.set(width / TWO_FOR_HALVING, 0, length / TWO_FOR_HALVING);
-        }
-        schematic.setOffset(offset);
-
-        @NotNull AxisAlignedBB region = AxisAlignedBB.fromBounds(minX, minY, minZ, maxX, maxY, maxZ);
-        //schematic::addEntity already does null checking and ignores null values.
-        world.getEntitiesWithinAABB(EntityHanging.class, region).forEach(schematic::addEntity);
-        world.getEntitiesWithinAABB(EntityMinecart.class, region).forEach(schematic::addEntity);
-
-        return schematic;
-    }
-
-    @NotNull
-    private static File getScanDirectory(@NotNull World world)
-    {
-        File minecolonies;
-        if (world.isRemote)
-        {
-            minecolonies = new File(Minecraft.getMinecraft().mcDataDir, "minecolonies/");
-        }
-        else
-        {
-            minecolonies = MinecraftServer.getServer().getFile("minecolonies/");
-        }
-        checkDirectory(minecolonies);
-
-        @NotNull File scans = new File(minecolonies, "scans/");
-        checkDirectory(scans);
-        return scans;
-    }
-
-    /**
-     * Creates a new tileEntity and formats its data for saving. A new tileEntity is needed to prevent changes to the real one.
-     *
-     * @param schematic  Schematic to add the TileEntity to.
-     * @param tileEntity The tile entity.
-     * @param minPos     The schematic min pos to subtract of the tile entity.
-     */
-    private static void saveTileEntity(@NotNull Schematic schematic, @Nullable TileEntity tileEntity, @NotNull BlockPos minPos)
-    {
-        if (tileEntity != null)
-        {
-            BlockPos newPos = tileEntity.getPos().subtract(minPos);
-            @NotNull NBTTagCompound tileEntityNBT = new NBTTagCompound();
-            tileEntity.writeToNBT(tileEntityNBT);
-
-            TileEntity newTileEntity = TileEntity.createAndLoadEntity(tileEntityNBT);
-            newTileEntity.setPos(newPos);
-            schematic.setTileEntity(newPos, newTileEntity);
-        }
-    }
-
-    private static void checkDirectory(@NotNull File directory)
-    {
-        if (!directory.exists() && !directory.mkdirs())
+        createScanDirectory(world);
+        String currentMillis = Long.toString(System.currentTimeMillis());
+        String fileName = "/../../../" + "minecolonies/scans/" + LanguageHandler.format("item.scepterSteel.scanFormat", "", currentMillis);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'fileName' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)